src/ripple/app/ledger/tests/common_ledger.cpp

```diff
@@ -435,28 +435,61 @@ trust(TestAccount& from, TestAccount const& issuer,
 void
 close_and_advance(Ledger::pointer& ledger, std::shared_ptr<Ledger const>& LCL)
 {
+    // This version of close_and_advance uses wall time.
+    using namespace std::chrono;
+    auto const epoch_offset = days(10957);  // 2000-01-01
+    auto const closeTime = time_point_cast<seconds>  // now
+        (system_clock::now() - epoch_offset).
+        time_since_epoch().count();
+
+    // Pass the time delta to the other close_and_advance.
+    close_and_advance (
+        ledger, LCL, closeTime - ledger->getParentCloseTimeNC());
+}
+
+void
+close_and_advance(Ledger::pointer& ledger,
+    std::shared_ptr<Ledger const>& LCL, std::uint32_t advSeconds)
+{
+    // An empty container of transactions to retry.
+    auto retries = LocalTxs::New();
+
+    // Let the most complicated version of close_and_advance do the lifting.
+    close_and_advance (ledger, LCL, advSeconds, retries);
+}
+
+void
+close_and_advance(Ledger::pointer& ledger, std::shared_ptr<Ledger const>& LCL,
```

Comment:
@vinniefalco, I hear your concerns, but that is the nature of a mock for unit testing.  And in this case I'm making the mock more sophisticated so it can test problems that I found using my external python tests.  Performing these tests using only `npm` tests will be slow, since Tickets must expire.  By using this interface I can both programmatically advance time and simulate transaction retries (both of which I need for certain test cases).